elasticsearch/connection/base.py

```diff
@@ -38,20 +47,54 @@ def __init__(
         use_ssl=False,
         url_prefix="",
         timeout=10,
+        headers=None,
+        http_compress=False,
+        cloud_id=None,
+        api_key=None,
         **kwargs
     ):
-        """
-        :arg host: hostname of the node (default: localhost)
-        :arg port: port to use (integer, default: 9200)
-        :arg url_prefix: optional url prefix for elasticsearch
-        :arg timeout: default timeout in seconds (float, default: 10)
-        """
+
+        if cloud_id:
+            try:
+                _, cloud_id = cloud_id.split(":")
+                parent_dn, es_uuid, _ = (
+                    binascii.a2b_base64(cloud_id.encode("utf-8")).decode("utf-8").split("$")
+                )
+            except ValueError:
+                raise ImproperlyConfigured("'cloud_id' is not properly formatted")
+
+            host = "%s.%s" % (es_uuid, parent_dn)
+            port = 9243
+            use_ssl = True
+            http_compress = True
```

Comment:
this doesn't seem ideal, it is (afaict) undocumented that we force compression on cloud connections and it is not always called for. `cloud_id` can also be used for local ECE installations where we might not want `http_compress`.

Maybe we can just set `http_compress` to `True` if it hasn't been set or just document that, when connecting to public cloud, it is a good idea to enable `http_compress`?

Reply:
This was added in a recent PR by me due to [this team issue](https://github.com/elastic/clients-team/issues/13) which recommended enabling compression, I'll add logic that doesn't override if the user sets an explicit value. I'll also add a tidbit in the cloud ID docs.